File: src/main/java/com/github/ojacquemart/apiauchan/restaurant/menu/Dish.java
package com.github.ojacquemart.apiauchan.restaurant.menu;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public class Dish implements Comparable<Dish> {
    private String name;
    private String price;

    @Override
    public int compareTo(Dish dishCompareTo) {
        int priceCompareTo = price.compareTo(dishCompareTo.getPrice());
        if (priceCompareTo != 0) {
            return priceCompareTo;
        }

        return name.compareTo(dishCompareTo.getName());
    }

    @Override
    public String toString() {
        return name + " " + price;
    }

}

File: src/main/kotlin/com/github/ojacquemart/api/restaurant/menu/Dish.kt
package com.github.ojacquemart.api.restaurant.menu

import com.github.ojacquemart.api.restaurant.menu.io.Writeable
import com.github.ojacquemart.api.restaurant.menu.io.WriterOption

data class Dish(val name: String, val price: String) : Comparable<Dish>, Writeable {

    override fun compareTo(other: Dish): Int {
        val priceCompareTo = price.compareTo(other.price)
        if (priceCompareTo != 0) {
            return priceCompareTo
        }

        return name.compareTo(other.name)
    }

    override fun write(options: List<WriterOption>): String {
        if (options.contains(WriterOption.DISH_PRICE)) {
            return "$name *($price)*"
        }

        return name
    }

}